options.c

```diff
@@ -2772,7 +2776,19 @@ void server_options(char **args, int *argc_p)
 		args[ac++] = arg;
 	}
 
-	if (io_timeout) {
+    if (max_map_size) {
```

Comment:
i think we will need a MAX_MAP_SIZE_DEFAULT define, and make this check be if (max_map_size != MAX_MAP_SIZE_DEFAULT)
otherwise we will be sending this option always, and the remove rsync may not understand it, so would break a lot of setups